### DISCUSSION TITLE
Azure Devops extension cannot access projects under a team

### DISCUSSION THREAD

**Discussion by TomaGagne** [vercel/community#3292] [Bug Report]
### Page to Investigate

https://vercel.com/docs/concepts/deployments/git/vercel-for-azure-pipelines

### Steps to Reproduce

1. Create a team
2. Follow the exact steps in the documentation of https://vercel.com/docs/concepts/deployments/git/vercel-for-azure-pipelines
3. At the step where you need to create your personal access token, choose the team you created.
4. For the Org ID, use the team ID as mentioned in the doc
5. For the Project ID, use the project ID of the project under your team as mentioned in the doc
6. When running the pipeline, you should see an error message saying `Error: Not authorized: Trying to access resource under scope <account username>. You must re-authenticate to this scope or use a token with access to this scope.`

However, the token used in the pipeline is set with access to the team, not the personal account. Why would it try to access my personal account scope?

If you decide to use a personal access token with your `Full Account `as scope, then the extension will create a brand new project under your personal account, which is not an expected behaviour since we are specifying the project ID and org ID. 

Here a screenshot of the pipeline output:
![image](https://github.com/vercel/community/assets/46379340/08ef3677-c1a5-491c-a5db-6a7fc9d21795)

Here is my pipeline yaml: 
![image](https://github.com/vercel/community/assets/46379340/8bb3263a-4de2-47ad-959a-1d81af2b932f)

Is there something missing in the doc to make the extension work for a Vercel team? Does the extension have a bug? 

Thanks for the help. Please ask questions if you need more information.

**Comment by amyegan** [comment]
Sorry you're having trouble with the extension. It sounds like the token isn't making it through to where it is needed. I'm sure you tried some of these ideas already, but here are a few things that might help get you unstuck:
- Try creating a new full account scope `VERCEL_TOKEN` and set it as a "variable" under the pipeline.
- If you have more than one pipeline, check that you have set the variable for each one that you are using.
- Check that the `VERCEL_TOKEN` value is coming through in `vercelToken` input as expected.

**Comment by TomaGagne** [comment]
Hi @amyegan, thanks for the answer. 

- When using a full account scope token, it creates a brand new project under my personal account and not under my team. 
- Currently, it is my only CD pipeline, so it is the only pipeline needing these values
- I even tried to put the token directly as a string in my yaml to be 100% sure the value was coming through and I still have the same output. 

Sadly, I'm still stuck with this issue. So I'll ask my team to use Github in the meantime.

Thanks again

**Comment by kamilkostrzewski1991** [comment]
Any updates? We are experiencing exactly the same problem. 

We tried:
- Use `VERCEL_ORG_ID` and `VERCEL_PROJECT_ID ` but it says can't find this project
- remove vercel.json
- add custom configurations


nothing helps. Problem appeared just like this. With no reason.

**Reply by amyegan** [reply]
Hey @kamilkostrzewski1991. Was the extension previously working correctly, or did you see `Error: Not authorized` as soon as you set it up?

**Reply by kamilkostrzewski1991** [reply]
I have seen unauthorized just after i used no-global token.

I fixed this error with adding flag `--scope=`

```
yaml
    - vercel pull  --yes --environment=production --token=$VERCEL_TOKEN --scope=kamilkostrzewskis-projects
    - APP_ENV=stage vercel  build --prod --token=$VERCEL_TOKEN --scope=kamilkostrzewskis-projects
    - vercel deploy --force --prebuilt --prod --token=$VERCEL_TOKEN --scope=kamilkostrzewskis-projects
```

It looks like Vercel started randomly using context of wrong Team. Forcing `--scope` was solution.
